src/onegov/api/views.py

```diff
@@ -10,24 +9,24 @@
 from onegov.api.models import ApiEndpoint, ApiException, AuthEndpoint, ApiKey
 from onegov.api.models import ApiEndpointCollection
 from onegov.api.models import ApiEndpointItem
-from onegov.api.token import get_token, jwt_decode
+from onegov.api.token import get_token, jwt_decode, try_get_encoded_token
 from onegov.core.security import Public
 
 
 def authenticate(request):
     try:
-        assert request.authorization[0].lower() == 'basic'
-        auth = b64decode(request.authorization[1].strip()).decode('utf-8')
-        auth, _ = auth.split(':', 1)
+        auth = try_get_encoded_token(request)
         data = jwt_decode(request, auth)
-        request.session.query(ApiKey).filter_by(id=data['id']).one()
     except jwt.ExpiredSignatureError as exception:
         raise HTTPUnauthorized() from exception
-    except NoResultFound as no_res:
-        raise HTTPClientError() from no_res
     except Exception as e:
         raise ApiException() from e
 
+    try:
+        request.browser_session.get(ApiKey, data['id'])
+    except NoResultFound as no_res:
+        raise HTTPClientError() from no_res
+
```

Comment:
```suggestion
    if request.session.query(ApiKey).get(data['id']) is None:
        raise HTTPClientError() from no_res

```
This is my bad `session.get` wasn't added to SQLAlchemy until 1.4, I forgot we were still on 1.3. So you will still need to use the query, but `get` on query is still faster than a filter. Also `get` will not raise an exception, but that's what we want

Reply:
Ah, I was actually curious what you mean by this. It didn't work, so I had to use`browser_session`, but that makes sense, I'll commit this.